packages/react-dom/src/__tests__/ReactDOMEventListener-test.js

```diff
@@ -189,9 +194,21 @@ describe('ReactDOMEventListener', () => {
         // The first call schedules a render of '1' into the 'Child'.
         // However, we're batching so it isn't flushed yet.
         expect(mock.mock.calls[0][0]).toBe('Child');
-        // The first call schedules a render of '2' into the 'Child'.
-        // We're still batching so it isn't flushed yet either.
-        expect(mock.mock.calls[1][0]).toBe('Child');
+        if (ReactFeatureFlags.enableModernEventSystem) {
+          // As we have two roots, it means we have two event listeners.
+          // This also means we enter the event batching phase twice,
+          // flushing the child to be 1.
+          // TODO: can we improve on this? microtasks seem to problematic
+          // (I tried that). We don't have any good way of knowing if
+          // another event will occur because another event handler
+          // might invoke stopPropagation() along the way. This means
+          // we over-flush.
```

Comment:
I have played around with this before and also did not find a good way to batch events. Micro tasks are also problematic indeed because they are exhausted before the next event listener is fired (https://jsfiddle.net/gm7r5fkp/) and this behaviour is different when you call the event listeners programatically.

But batched mode (is this still a thing?) or concurrent mode should fix this, right?

Reply:
I spoke in depth to @sebmarkbage last nigtht about this and we settled on the fact we'll just have to put up with over-flushing till we can get around to tuning this in the future with some capability in the scheduler (which might require platform features being added).

Batched mode is still definitely a thing in concurrent mode though, and it's still something you'd ideally like to be automatic for the user when calling events.